selftests/unit/test_utils_partition.py

```diff
@@ -45,12 +37,8 @@ class TestPartition(unittest.TestCase):
 
     @unittest.skipIf(missing_binary('mkfs.ext2'),
                      "mkfs.ext2 is required for these tests to run.")
-    @unittest.skipIf(missing_binary('sudo'),
+    @unittest.skipIf(not process.can_sudo(),
```

Comment:
There is no 100% way unless you try executing the command. The thing is that most of the superuser programs work well without sudo to get the basic info. Commands like `mount`, `mkfs.ext2 -V`, `lvs` ... work well, report 0 but when you try actually mounting, formatting or getting the volumes, it fails.

This is why I think this solution is the best as it supports well known configuration. If you know a better way, please let me know.